idaes/power_generation/unit_models/boiler_heat_exchanger.py

```diff
@@ -78,135 +92,279 @@ class DeltaTMethod(Enum):
+    HeatExchanger configuration option.
+    """
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return (dT1[t] - dT2[t]) / log(dT1[t] / dT2[t])
+
+
+def delta_temperature_amtd_callback(b):
+    """
+    This is a callback for a temperature difference expression to calculate
+    :math:`\Delta T` in the heat exchanger model using arithmetic-mean
+    temperature difference (AMTD).  It can be supplied to
+    "delta_temperature_callback" HeatExchanger configuration option.
+    """
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return (dT1[t] + dT2[t]) * 0.5
+
+
+def delta_temperature_underwood_callback(b):
+    """
+    This is a callback for a temperature difference expression to calculate
+    :math:`\Delta T` in the heat exchanger model using log-mean temperature
+    difference (LMTD) approximation given by Underwood (1970).  It can be
+    supplied to "delta_temperature_callback" HeatExchanger configuration option.
+    This uses a cube root function that works with negative numbers returning
+    the real negative root. This should always evaluate successfully.
+    """
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+    temp_units = pyunits.get_units(dT1)
+
+    # external function that ruturns the real root, for the cuberoot of negitive
+    # numbers, so it will return without error for positive and negitive dT.
+    b.cbrt = ExternalFunction(
+        library=functions_lib(), function="cbrt", arg_units=[temp_units]
+    )
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return ((b.cbrt(dT1[t]) + b.cbrt(dT2[t])) / 2.0) ** 3 * temp_units
+
+
+def delta_temperature_underwood_tune_callback(b):
+    """
+    This is a callback for a temperature difference expression to calculate
+    :math:`\Delta T` in the heat exchanger model using log-mean temperature
+    difference (LMTD) approximation given by Underwood (1970).  It can be
+    supplied to "delta_temperature_callback" HeatExchanger configuration option.
+    This uses a cube root function that works with negative numbers returning
+    the real negative root. This should always evaluate successfully.
+    """
+    temp_units = pyunits.get_units(b.deltaT_1)
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+    b.lmtd_param_c1 = Var(initialize=0.3241)
+    b.lmtd_param_c2 = Var(initialize=1.99996)
+    b.lmtd_param_c1.fix()
+    b.lmtd_param_c2.fix()
+    c1 = b.lmtd_param_c1
+    c2 = b.lmtd_param_c2
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return (((dT1[t] / temp_units) ** c1 + (dT2[t] / temp_units) ** c1) / c2) ** (
+            1 / c1
+        ) * temp_units
+
+
 @declare_process_block_class("BoilerHeatExchanger")
 class BoilerHeatExchangerData(UnitModelBlockData):
     """
     Standard Heat Exchanger Unit Model Class
     """
+
     CONFIG = ConfigBlock()
```

Comment:
Is there any reason you are not inheriting from the Config Block in the base class here? Notably, the base Config already has `dynamic` and `has_holdup` defined for you (as they are expected arguments).

Reply:
I didn't write this model, and I didn't want to get into rewriting it.  Overall it would be nice if it inherited the latest 0D heat exchanger model, but I'm not up for that much change right now.  In the future we should probably make some of these heat transfer and pressure drop correlations available in the generic model and do a serious update of the boiler heat exchanger model.